src/plots/cartesian/layout_attributes.js

```diff
@@ -387,6 +387,14 @@ module.exports = {
             'plotted on'
         ].join(' ')
     },
+    spikesnap: {
+        valType: 'enumerated',
+        values: ['data', 'cursor'],
+        dflt: 'data',
+        role: 'style',
```

Comment:
Though always debatable, all the new attributes you added should be `role: 'info'` as they do affect the _behavior_ of the graph, not just the aesthetics.

Reply:
Thanks! Changed it for `hoverdistance` and `spikedistance`. Didn't change it for the spikesnap because all other axis properties (like `showspikes` and `spikemode`) have role `style`.